### ISSUE TITLE
linter: typescript/no-deprecation should report usage of deprecated fields in types.

### ISSUE THREAD

**Issue by refparo** [oxc-project/oxc#22779]
This code should be disallowed by typescript/no-deprecation:

```ts
interface Something {
  /**
   * @deprecated
   */
  field: number;
}

function bar(_options: Something) {
}

bar({ field: 0 });
```

And equivalently, this code:

```tsx
interface SomethingProps {
  /**
   * @deprecated
   */
  field: number;
}

function Foo(props: SomethingProps) {
}

const jsx = <Foo field={0} />;
```

**Comment by camc314** [comment]
Hmm i thought I added support for this here: https://github.com/oxc-project/tsgolint/pull/944

**Comment by camc314** [comment]
Already implemented in https://github.com/oxc-project/tsgolint/pull/944

(i need to do a tsgolint release)
